docs/scripts/util/checkLink.ts

```diff
@@ -100,7 +100,7 @@ export async function checkLink(
               ...linkData,
               statusCode,
             })
-          )?.statusCode || statusCode;
+          )?.statusCode ?? statusCode;
```

Comment:
This is the major change of this new PR 👿

Reply:
Can you please explain what this change does?

Reply:
Yes. 

If a link is good we would return a `statusCode: 0` ([code](https://github.com/aws-amplify/amplify-ui/blob/main/docs/scripts/util/checkLink.ts#L91)). The problems was, the redirected link got a `0`, and it was `308` before redirected, so `0 || 308` and we got `308`.

We should have been able to filter out the `308` errors when printing errors ([code](https://github.com/aws-amplify/amplify-ui/blob/main/docs/scripts/link-checker-puppeteer.ts#L20)), but since the `SITE_URL` Github Env Variable was not set to be `localhost:3000`, it was not filtered out. 

That's why we got those internal 308 errors which failed the build.

Since we've changed the Github Env Variable, the problem should be resolved.

So, it's actually not necessary to change `||` to `??`.